### ISSUE TITLE
References to enumerations within a class named Bool are not properly handled

### ISSUE THREAD

**Issue by vilkarn** [doxygen/doxygen#10190]
In a c++ header file with a class named `Bool` references are not linked.

```{.cpp}
/// working class example
///
/// \ref Boole::ExampleEnum<br>
/// \ref Boole::True
class Boole
{
public:
/// test enum
enum ExampleEnum
{
True, ///< true
Unknown, ///< unknown
False  ///< false
};
};

/// broken class example
///
/// \ref Bool::ExampleEnum<br>
/// \ref Bool::True
class Bool
{
public:
/// test enum
enum ExampleEnum
{
True, ///< true
Unknown, ///< unknown
False  ///< false
};
};
```

**Expected behavior**
It is expected that Bool would be handled the same way other class names are handled.

**To Reproduce**
Place the above example code in a header file with the a default doxygen configuration file and inspect the results to see that the Boole class links work, but the Bool class links are not resolved.

**Version**
1.9.6

**Comment by albert-github** [comment]
With the `Bool` class I get:

![image](https://github.com/doxygen/doxygen/assets/5223533/8306f133-fd32-4ce4-9aa7-b99f44ff57d7)


and this looks 100% similar to the picture with the `Boole` class.

- Which links don't work?
- Did I miss something?

**Comment by vilkarn** [comment]
It looks like there was a piece I was missing in my original description.  We have a tag file to another set of documentation that typedefs `Bool`.  I was able to recreate the problem by putting the following into a tag file and assigning it to `TAGFILES`.

```{.xml}
  <compound kind="file">
    <name>myTypes.h</name>
    <path>/path/to/file</path>
    <filename>myTypes_8h.html</filename>
    <namespace>myTypes</namespace>
    <member kind="typedef">
      <type>bool</type>
      <name>Bool</name>
      <anchorfile>namespacemyTypes.html</anchorfile>
      <anchor>123456789</anchor>
      <arglist></arglist>
    </member>
  </compound>
```

In the broken example the links to the type and value are not connected.
![image](https://github.com/doxygen/doxygen/assets/23289723/4ffe7e00-e9b9-4697-8d2f-3806ab33c960)

In the working example I get the same result you saw.
![image](https://github.com/doxygen/doxygen/assets/23289723/487ae296-4cb5-407d-b947-29edb96f5d9f)

**Comment by albert-github** [comment]
You now specify a tag file but before you wrote "a default doxygen configuration file so to have the same code and be ablle to reproduce the problem:
- Can you please attach a, small, self contained example (source+configuration file in a, compressed, tar or zip file) that allows us to reproduce the problem? Please don't add external links as they might not be persistent.

**Comment by albert-github** [comment]
Most likely cause looks like to be that you didn't copy enough into the `TAGFILES` i.e. heading of the tag file or the information about the `Bool` class and its members.

**Comment by vilkarn** [comment]
I attached example code below.  In proj1 the default file was modified to EXTRACT_ALL and GENERATE_TAGFILE was set.  In proj2 the default config was only modified to set TAGFILES to the relative path for the file produced by proj1.

[test_dox.zip](https://github.com/doxygen/doxygen/files/12134565/test_dox.zip)

**Comment by albert-github** [comment]
With the 1.9.6, 1.9.7 and the current master version (1.9.8 (7df783537e4ca190389340a9a0d93fc1a6f35580)) I can reproduce the problem. (the warning `test.h:19: warning: unable to resolve reference to 'Bool::ExampleEnum<br>' for \ref command` is a bit strange with the `<br>, but this is due to the template handling and has no influence on the problem.

In the `proj1` we  see in the `MyTypes.h`:
```
namespace myTypes
{
typedef bool Bool;
}
```
and in the `proj2` (file `types.h` we see just `class Bool` with its members.

In the tag file from `proj1` we see:
```
<?xml version='1.0' encoding='UTF-8' standalone='yes' ?>
<tagfile doxygen_version="1.9.8" doxygen_gitid="7df783537e4ca190389340a9a0d93fc1a6f35580*">
  <compound kind="file">
    <name>myTypes.h</name>
    <path></path>
    <filename>myTypes_8h.html</filename>
    <namespace>myTypes</namespace>
    <member kind="typedef">
      <type>bool</type>
      <name>Bool</name>
      <anchorfile>namespacemyTypes.html</anchorfile>
      <anchor>a17b3d5d06ba1f166c72c74bbdbedf4a2</anchor>
      <arglist></arglist>
    </member>
  </compound>
  <compound kind="namespace">
    <name>myTypes</name>
    <filename>namespacemyTypes.html</filename>
    <member kind="typedef">
      <type>bool</type>
      <name>Bool</name>
      <anchorfile>namespacemyTypes.html</anchorfile>
      <anchor>a17b3d5d06ba1f166c72c74bbdbedf4a2</anchor>
      <arglist></arglist>
    </member>
  </compound>
</tagfile>
```

so it looks like the `compound` `file` has a `namespace` and also a separate `typedef` where the `typedef` is not  related to the `namespace`.

I think the `file` only has a namespace.
@doxygen is this correct and how to correct?

**Comment by albert-github** [comment]
I've just pushed a proposed patch, pull request #10198

**Comment by albert-github** [comment]
Code has been integrated in master on GitHub (please don't close the issue as this will be done at the moment of an official release).

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.9.8.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
